source/common/http/conn_manager_impl.cc

```diff
@@ -2004,6 +2014,12 @@ bool ConnectionManagerImpl::ActiveStream::createFilterChain() {
   }
   bool upgrade_rejected = false;
   auto upgrade = request_headers_ ? request_headers_->Upgrade() : nullptr;
+  // Treat CONNECT requests as a special upgrade case.
+  if (!upgrade && request_headers_ && request_headers_->Method() &&
```

Comment:
Is it possible to have headers but not method?

Reply:
I'm fine either way. I thought this was before filters run but it's not a big deal either way.